### ISSUE TITLE
[sklearn#34327] Creating pipelines with default arguments for methods in steps

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#335]
**Mirrored from [scikit-learn#34327](https://github.com/scikit-learn/scikit-learn/issues/34327)**

Original title: Creating pipelines with default arguments for methods in steps
Original labels: New Feature, Needs Decision, Metadata Routing
Created: 2026-06-17T18:13:48Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

### Describe the workflow you want to enable

Sometimes, one might want to create pipelines where some steps will have additional non-data arguments in methods like `.transform()`.

This can be done through metadata routing, but: (a) it's not easy to desgin pipelines like this, particularly when they are created programmatically and one wishes to use a given transformer class multiple times with different parameters; (b) it requires passing the additional parameters manually when calling methods, instead of being there by default:

```python
from sklearn import make_pipeline
from sklearn.linear_model import LogisticRegression
from optbinning import BinningProcess

pipeline = make_pipeline(
    BinningProcess().set_transform_request(metric_missing=True),
    LogisticRegression(),
)
pipeline.fit(
    X,y
).predict(
    X,
    metric_missing="empirical" # <- want this 'on' by default
)
```

One could also subclass objects or overwrite methods with `partial`, but neither is convenient.

Would be useful if `Pipeline()` could offer additional fixed arguments to pass to methods, like this:
```python
Pipeline([
    "binner",
    BinningProcess(),
    {"transform": {"metric_missing": "empirical"}}
])
```

### Describe your proposed solution

Accepting a third argument in the tuples that go into `Pipeline`. Maybe also `ColumnTransformer`.

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/88 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
